File: 2/2000s/20s/2091/RemovingMinimumAndMaximumFromArray.py
# https://leetcode.com/problems/removing-minimum-and-maximum-from-array/description/?envType=daily-question&envId=2026-08-30

class Solution:
    def minimumDeletions(self, nums: List[int]) -> int:
        lengthOfNums=len(nums)
        maxElement=-9874563211
        minElement=9874563211
        delCountForMaxFromLeft=0
        delCountForMaxFromRight=0
        finalDelCount=9874563211
        indexForMinElement=-1
        for i in range(0, lengthOfNums, 1):
            maxElement=max(maxElement, nums[i])
            if nums[i]<=minElement:
                minElement=min(minElement, nums[i])
                indexForMinElement=i
        for i in range(0, lengthOfNums, 1):
            if nums[i]==maxElement:
                delCountForMaxFromLeft+=1
                if indexForMinElement<=i:
                    finalDelCount=min(finalDelCount, delCountForMaxFromLeft)
                if indexForMinElement>i:
                    finalDelCount=min(finalDelCount, indexForMinElement+1)
                    finalDelCount=min(finalDelCount, delCountForMaxFromLeft+(lengthOfNums-indexForMinElement))
                break
            else:
                delCountForMaxFromLeft+=1
        for i in range(lengthOfNums-1, -1, -1):
            if nums[i]==maxElement:
                delCountForMaxFromRight+=1
                if indexForMinElement>=i:
                    finalDelCount=min(finalDelCount, delCountForMaxFromRight)
                if indexForMinElement<i:
                    finalDelCount=min(finalDelCount, lengthOfNums-indexForMinElement)
                    finalDelCount=min(finalDelCount, delCountForMaxFromRight+(indexForMinElement+1))
                break
            else:
                delCountForMaxFromRight+=1
        return finalDelCount

File: 2/2000s/20s/2091/RemovingMinimumAndMaximumFromArray.cpp
/*
https://leetcode.com/problems/removing-minimum-and-maximum-from-array/description/?envType=daily-question&envId=2026-08-30
*/

class Solution {
public:
    int minimumDeletions(vector<int>& nums) {
        int sz=nums.size(), maxx=INT_MIN, minn=INT_MAX, countLForMax=0, countRForMax=0, fnCount=INT_MAX;
        int idxForMinn=-1;
        for(int i=0; i<sz; ++i){
            maxx=max(maxx, nums[i]);
            if(nums[i]<=minn){
                minn=min(minn, nums[i]);
                idxForMinn=i;
            }
        }
        for(int i=0; i<sz; ++i){
            if(nums[i]==maxx){
                countLForMax++;
                if(idxForMinn<=i){
                    fnCount=min(fnCount, countLForMax);
                }
                if(idxForMinn>i){
                    fnCount=min(fnCount, idxForMinn+1);
                    fnCount=min(fnCount, countLForMax+(sz-idxForMinn));
                }
                break;
            } else{
                countLForMax++;
            }
        }
        for(int i=sz-1; i>=0; --i){
            if(nums[i]==maxx){
                countRForMax++;
                if(idxForMinn>=i){
                    fnCount=min(fnCount, countRForMax);
                }
                if(idxForMinn<i){
                    fnCount=min(fnCount, sz-idxForMinn);
                    fnCount=min(fnCount, countRForMax+(idxForMinn+1));
                }
                break;
            } else{
                countRForMax++;
            }
        }
        return fnCount;
    }
};